core/src/test/resources/test_workflow.json

```diff
@@ -0,0 +1,256 @@
+            "x-correlation-id": "${workflow.correlationId}"
+          }
+        }
+      },
+      "type": "HTTP",
+      "startDelay": 0,
+      "optional": false
+    },
+    "rateLimitPerFrequency": 0,
+    "rateLimitFrequencyInSeconds": 1,
+    "loggingEnabled": true,
+    "taskDefinition": {
+      "present": false
+    },
+    "queueWaitTime": 479,
+    "taskStatus": "COMPLETED"
+  }],
+  "input": {
+    "context": {},
+    "headers": {
+      "header1": "Header1"
+    },
+    "body": {}
+  },
+  "output": {
+    "response": {
+      "headers": {
+        "header1": "Header1"
+      },
+      "reasonPhrase": "OK",
+      "body": {
+        "hello": "world"
+      },
+      "statusCode": 200
+    }
+  },
+  "workflowType": "ConditionalMode",
+  "version": 3,
+  "correlationId": "null",
+  "schemaVersion": 2,
+  "workflowDefinition": {
+    "createTime": 1550686558442,
+    "name": "orchestrator_ConditionalMode",
+    "description": "...",
+    "version": 3,
+    "tasks": [
+      {
+        "name": "orchestrator_Default",
+        "taskReferenceName": "Conditional01",
+        "description": "...",
+        "inputParameters": {
+          "caseValueParam": "${workflow.input.body.mode}"
+        },
+        "type": "DECISION",
+        "caseValueParam": "caseValueParam",
+        "decisionCases": {
+          "01": [
+            {
+              "name": "orchestrator_Default",
+              "taskReferenceName": "Request01",
+              "inputParameters": {
+                "http_request": {
+                  "uri": "http://www.mocky.io/v2/5c6d95ff370000530afa34e8",
+                  "method": "GET",
+                  "headers": {
+                    "x-workflow-id": "${workflow.workflowId}",
+                    "x-correlation-id": "${workflow.correlationId}"
+                  }
+                }
+              },
+              "type": "HTTP",
+              "startDelay": 0,
+              "optional": false
+            }
+          ]
+        },
+        "defaultCase": [
+          {
+            "name": "orchestrator_Default",
+            "taskReferenceName": "Request02",
+            "inputParameters": {
+              "http_request": {
+                "uri": "http://www.mocky.io/v2/5c6d95ff370000530afa34e8",
+                "method": "GET",
+                "headers": {
+                  "x-workflow-id": "${workflow.workflowId}",
+                  "x-correlation-id": "${workflow.correlationId}"
+                }
+              }
+            },
+            "type": "HTTP",
+            "startDelay": 0,
+            "optional": false
+          }
+        ],
+        "startDelay": 0,
+        "optional": false
+      }
+    ],
+    "schemaVersion": 2,
+    "restartable": true,
+    "workflowStatusListenerEnabled": false,
+    "obfuscationFields": {
+      "workflowFields": "output.response.body",
+      "taskFields": "Request02.inputData.http_request.headers.x-workflow-id"
```

Comment:
I see the fields are comma separated. And the implementation doesn't have a trim, to avoid accidental spaces after comma. It might be best to have json arrays of tokens. Eg: 

```json
"obfuscationFields": {
      "workflowFields": [
            "output.response.body",
            "output.someKey"
      ],
      "taskFields": [
            "Request02.inputData.http_request.headers.x-workflow-id",
            "Request04.outputData.someValue"
      ]
 }
```

Reply:
Actually at the first attempt to model this new field I opted for a Map<String, <List<String>> but I got some errors with gradle's protogen task, it seems the lib used has some struggles with nested types. When I switched to a Map<String, String> it worked normally.
Maybe I missed some additional configuration or if you know there's a workaround for this I'd be glad to change the field type, otherwise I'll add a trim to the implementation.